TRAMS-API/Controllers/ProjectsController.cs

```diff
@@ -21,23 +23,104 @@ namespace API.Controllers
     [Produces("application/json")]
     [ProducesResponseType(typeof(string), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(string), StatusCodes.Status429TooManyRequests)]
+    [ProducesResponseType(typeof(string), StatusCodes.Status502BadGateway)]
     [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
     public class ProjectsController : Controller
     {
         private readonly IProjectsRepository _projectsRepository;
         private readonly IAcademiesRepository _academiesRepository;
         private readonly ITrustsRepository _trustsRepository;
-        private readonly IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> _mapper;
+        private readonly IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> _postProjectsMapper;
+        private readonly IMapper<GetProjectsD365Model, GetProjectsResponseModel> _getProjectsMapper;
+        private readonly IMapper<AcademyTransfersProjectAcademy, 
+                                 Models.Upstream.Response.GetProjectsAcademyResponseModel> _getProjectAcademyMapper;
+        private readonly IRepositoryErrorResultHandler _repositoryErrorHandler;
 
         public ProjectsController(IProjectsRepository projectsRepository,
                                   IAcademiesRepository academiesRepository,
                                   ITrustsRepository trustsRepository,
-                                  IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> mapper)
+                                  IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> postProjectsMapper,
+                                  IMapper<GetProjectsD365Model, GetProjectsResponseModel> getProjectsMapper,
+                                  IMapper<AcademyTransfersProjectAcademy, Models.Upstream.Response.GetProjectsAcademyResponseModel> getProjectAcademyMapper,
+                                  IRepositoryErrorResultHandler repositoryErrorHandler)
         {
             _projectsRepository = projectsRepository;
             _academiesRepository = academiesRepository;
             _trustsRepository = trustsRepository;
-            _mapper = mapper;
+            _postProjectsMapper = postProjectsMapper;
+            _getProjectsMapper = getProjectsMapper;
+            _getProjectAcademyMapper = getProjectAcademyMapper;
+            _repositoryErrorHandler = repositoryErrorHandler;
+        }
+
+        /// <summary>
+        /// Gets an Academy Transfers project by its ID
+        /// </summary>
+        /// <param name="id">The ID of the project</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("/projects/{id}")]
+        [ProducesResponseType(typeof(GetProjectsResponseModel), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetTrustById(Guid id)
```

Comment:
Error: This method is getting a project not a trust.  Whilst not visible to the consumer but feel it should be renamed.

Reply:
Good spot, will rename